src/java.base/unix/native/libjimage/osSupport_unix.cpp

```diff
@@ -38,7 +38,7 @@
  * Return the file descriptor.
  */
 jint osSupport::openReadOnly(const char *path) {
-    return ::open(path, 0);
+    return ::open(path, O_CLOEXEC);
```

Comment:
This is okay but I think it would be useful to know why this one place needs O_CLOEXEC and not others.

Reply:
Probably others too, if we care about inheriting read handles to child.

The background is that I am playing with a new test which checks that the VM has no open inheritable file descriptors. It is supposed to replace some specialized tests which are maintenance-intensive. I am still tuning the test, since at the moment it turns out too many false positives.

Anyway, this is the very first descriptor the VM opens, therefore it triggered my test. I thought since there is a sibling fix for windows, a similar fix for posix would be symmetric.

If you think this is not worth fixing, or we should fix all occurrences in one swoop, that is fine too.

Reply:
On Unix systems, the JDK has always relied on the Runtime.exec implementation to close the file descriptors. On Windows the inheritance has to be disabled in the parent. So if the gtest launcher is forking directly then we may have a bit of whack-a-mole to change all the places that open file descriptors.

Reply:
lib/modules is opened/mapped early in the startup so I assume this is why this one shows up quickly. Adding O_CLOEXEC everywhere is open, up you if you want to do that or switch it to Runtime.exec.